.github/workflows/ci.yml

```diff
@@ -682,6 +682,17 @@ jobs:
             - default
             - hdp3
             # TODO: config-apache-hive3
+          ignore exclusion if:
+            # Do not use this property outside of the matrix configuration.
+            #
+            # This is added to all matrix entries so they may be conditionally
+            # excluded by adding them to the excludes list with a GHA expression
+            # for this property.
+            # - If the expression evaluates to true, it will never match the a
+            #   actual value of the property, and will therefore not be excluded.
+            # - If the expression evaluates to false, it will match the actual
+            #   value of the property, and the exclusion will apply normally.
+            - false # intentionally a boolean instead of string like other matrix keys to make it easier to evaluate expressions
```

Comment:
I changed this to have the same type here and in places where we evaluate expressions, like this:
```
              ignore exclusion if: >-
                ${{ needs.build-pt.outputs.have_databricks_secrets == 'true' }}
```

Without this, after outputting the matrix as JSON, it seemed like items from the `exclude` sections were not matched with this default.

Check the example JSON matrix here: https://github.com/trinodb/trino/actions/runs/3253652165/jobs/5341105651 where all types are strings.

Reply:
In this run: https://github.com/trinodb/trino/actions/runs/3257518379/jobs/5348831674
we have:
![image](https://user-images.githubusercontent.com/795177/196148157-c0a6c48a-4202-4f29-9597-4fd19584a33f.png)
but the matrix yaml gets saved as:
![image](https://user-images.githubusercontent.com/795177/196148233-df4bcb85-6e26-4b8d-ad54-f63ba8e1ee15.png)

Reply:
We moved some steps from the `pt` job to `build-pt` but we're referring to outputs using the job id, instead of changing it to step id.

Currently, we use:
```
needs.build-pt.outputs.have_azure_secrets
```
but we should be using
```
needs.check-secrets.outputs.have_azure_secrets
```

Reply:
https://github.com/trinodb/trino/pull/14662